3scale/tasks/main.yml

```diff
@@ -0,0 +1,149 @@
+---
+- name: Create Secret in target_namespace
+  kubernetes.core.k8s:
+    state: present
+    namespace: "{{ target_namespace }}"
+    definition:
+      apiVersion: v1
+      kind: Secret
+      metadata:
+        name: aws-auth
+      stringData:
+        AWS_ACCESS_KEY_ID: "$(echo $ACCESS_CREDS | cut -f 1)"
+        AWS_SECRET_ACCESS_KEY: "$(echo $ACCESS_CREDS | cut -f 2)"
+        AWS_BUCKET: "$S3_BUCKET"
+        AWS_REGION: "$REGION"
+      type: Opaque
+- name: Create APIManager in target_namespace
+  kubernetes.core.k8s:
+    state: present
+    namespace: "{{ target_namespace }}"
+    definition:
+      apiVersion: apps.3scale.net/v1alpha1
+      kind: APIManager
+      metadata:
+        name: apimanager-sample
+      spec:
+        wildcardDomain: 3scale.${CLUSTER_BASE_URL}
+        system:
+          fileStorage:
+            simpleStorageService:
+              configurationSecretRef:
+                name: aws-auth
+- name: Create APICast for Staging in target_namespace
+  kubernetes.core.k8s:
+    state: present
+    namespace: "{{ target_namespace }}"
+    definition:
+      apiVersion: apps.3scale.net/v1alpha1
+      kind: APIcast
+      metadata:
+        name: example-apicast-staging
+      spec:
+        deploymentEnvironment: staging
+        configurationLoadMode: lazy
+        adminPortalCredentialsRef:
+          name: 3scaleportal
+- name: Create APICast for Production in target_namespace
+  kubernetes.core.k8s:
+    state: present
+    namespace: "{{ target_namespace }}"
+    definition:
+      apiVersion: apps.3scale.net/v1alpha1
+      kind: APIcast
+      metadata:
+        name: example-apicast-production
+      spec:
+        adminPortalCredentialsRef:
+          name: 3scaleportal
+- name: Install and report - APIManager
+  block:
+    - name: Wait for installation of APIManager
+      kubernetes.core.k8s_info:
+        api_version: apps.3scale.net/v1alpha1
+        kind: APIManager
+        name: apimanager-sample
+        namespace: "{{ target_namespace }}"
+      register: result
+      until: result.resources[0].status.conditions.status == "True" and result.resources[0].status.conditions.type. == "Available"
```

Comment:
It if follows the conditions model, you can use `wait_condition` instead of this loop. The loop is only necessary if conditions are not followed.

```yaml
wait: true
wait_condition:
    status: "True"
    type: "Available"
```